csrc/host_ir/jit.cpp

```diff
@@ -325,40 +536,41 @@ void compile(HostIrJitImpl* pimpl) {
       error);
 
   // Add the module to the JIT
-  throwIfError(pimpl->jit->addIRModule(
+  throwIfError(jit_->addIRModule(
       llvm::orc::ThreadSafeModule(std::move(mod), std::move(ctx))));
 
   // Look up the main function
-  auto main_func_addr = throwIfError(pimpl->jit->lookup(kMainFuncName));
+  auto main_func_addr = throwIfError(jit_->lookup(kMainFuncName));
   using main_func_ptr_t = void (*)(const void**, void**);
   auto main_func_ptr =
       reinterpret_cast<main_func_ptr_t>(main_func_addr.getValue());
-  pimpl->main_func = main_func_ptr;
+  main_func_ = main_func_ptr;
```

Comment:
```
  main_func_ = reinterpret_cast<main_func_ptr_t>(main_func_addr.getValue());
```

Reply:
I know I suggested main_func_ be a std::function in a previous PR. Now that I see how it's used in practice, I think std::function is an overkill. We are not going to use any feature (e.g. lambda and partial) in std::function other than as a function pointer.